sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/rules.scala

```diff
@@ -402,6 +403,22 @@ case class PreprocessTableInsertion(conf: SQLConf) extends Rule[LogicalPlan] {
           s"including ${staticPartCols.size} partition column(s) having constant value(s).")
     }
 
+    val partitionsTrackedByCatalog = conf.manageFilesourcePartitions &&
```

Comment:
We shouldn't check the config. The flag is decided by the config that was used to create this table, so `catalogTable.get.tracksPartitionsInCatalog` is good enough.

Reply:
manageFilesourcePartitions==false && tracksPartitionsInCatalog=true
`InsertIntoHadoopFsRelationCommand` does not call `catalog.listPartitions` and `AlterTableAddPartitionCommand`, so it does not check the partition value is empty.

```sql
set spark.sql.hive.manageFilesourcePartitions=false;
insert ovwriter table a partition(d='1',h='') select 1;
```
currently this sql execution will not fail.

Reply:
I mean, `conf.manageFilesourcePartitions` is used to create the table. When we need to read/alter table, we should respect `table.tracksPartitionsInCatalog`.